### PR TITLE
Skip using increment if nodes are adding to the end of some other node

### LINKED ISSUE #78381: Move multiple nodes to the end of their parent by dragging will actually move some of them to the beginning
### Godot version

4.1-beta2

### System information

Godot v4.1.beta2.mono - Ubuntu 23.04 23.04 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 6GB (nvidia; 515.105.01) - Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz (8 Threads)

### Issue description

If multiple nodes are selected in the editor and drag them to the end of their parent in the scene tree, every node except the first one will be moved to the beginning of its parent.

Right click selected nodes and select move down will give expected result

https://github.com/godotengine/godot/assets/6362922/beab3a3d-0971-448f-9970-afe564d8b7f6

### Steps to reproduce

Create a parent node
Add few children nodes
Select some nodes and drag them to the end of the parent node

### Minimal reproduction project

[NodeArrangementReproduce.zip](https://github.com/godotengine/godot/files/11780108/NodeArrangementReproduce.zip)

### PR DESCRIPTION
**Pull request by zhhrozhh**
Think we don't need to check the inc for nodes that reparent to the end of other node, as adding a positive inc to -1 will make the position non-negative.

tested scenarios:

https://github.com/godotengine/godot/assets/6362922/4923f861-cb7d-4012-8f3b-8629c4a6ed4e

 
Fixes #78381

### CODE REVIEW

**Review by kleonc — CHANGES_REQUESTED** [review]

**Comment by kleonc** [comment]
@zhhrozhh Note that a single commit is preferred, so please clean up the commit log (fixup/squash + I'd suggest rewording the commit title to be simpler / more general, e.g. something like _"Fix drag-dropping nodes at the end of children list"_). [See the docs for how-to](https://docs.godotengine.org/en/4.0/contributing/workflow/pr_workflow.html#the-interactive-rebase).

**Review by kleonc — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:
